platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -694,6 +692,35 @@ private Response showLoginPage(
         LOGGER.debug("Received Passive & PKI AuthnRequest.");
         org.opensaml.saml.saml2.core.Response samlpResponse;
         try {
+
+          // Find binding supported by SP and change template
+          String assertionConsumerServiceBinding =
+              ResponseCreator.getAssertionConsumerServiceBinding(
+                  authnRequest, getServiceProvidersMap());
+
+          if (HTTP_POST_BINDING.equals(assertionConsumerServiceBinding)) {
+            binding =
+                new PostBinding(
+                    systemCrypto,
+                    getServiceProvidersMap(),
+                    getPresignPlugins(),
+                    spMetadata,
+                    SUPPORTED_BINDINGS);
+            template = submitForm;
+          } else if (HTTP_REDIRECT_BINDING.equals(assertionConsumerServiceBinding)) {
+            binding =
+                new RedirectBinding(
+                    systemCrypto,
+                    getServiceProvidersMap(),
+                    getPresignPlugins(),
+                    spMetadata,
+                    SUPPORTED_BINDINGS);
+            template = redirectPage;
+          } else {
+            throw new IdpException(
+                new UnsupportedOperationException("Must use HTTP POST or Redirect bindings."));
+          }
+
```

Comment:
❓ I wonder if we're getting close to the point where we should have some separate service that determines the correct template and creates the `Binding` object based on the input. We have very similar code blocks throughout this class but they have very small differences which keep us from reusing the logic.

Reply:
I'm doing that in a separate commit right now 😄

Reply:
Working on this a bit more, idk if I should do this in this PR and on 2.11.x. @bakejeyner's PRs for extracting the plugin and the sp metadata parser were both only on master so I assume we don't want to mess with the IdP code on 2.11.x. I'm going to opt to create another ticket or I can do the work under the same ticket but once my PRs get merged.

Reply:
I'm in favor of a separate ticket 👍